truffle/src/main/java/org/jruby/truffle/core/CoreLibrary.java

```diff
@@ -278,7 +279,7 @@
     private final GlobalVariables globalVariables;
     private final DynamicObject mainObject;
     private final DynamicObject nilObject;
-    private final DynamicObject rubiniusUndefined;
+    private final Object rubiniusUndefined;
```

Comment:
Yes, some cleanup can be done after, since we can get rid of this whole field and guard.